app/clients/cbc_proxy.py

```diff
@@ -100,28 +108,42 @@ def cancel_broadcast(
     ):
         pass
 
-    def _invoke_lambda(self, payload):
-        if not self.lambda_name:
-            current_app.logger.warning(
-                '{self.__class__.__name__} tried to send {payload} but cbc proxy aws env vars not set'
-            )
-            return
-
+    def _invoke_lambda_with_failover(self, payload):
         payload_bytes = bytes(json.dumps(payload), encoding='utf8')
+        result = self._invoke_lambda(self.lambda_name, payload_bytes)
+
+        if not result:
+            failover_result = self._invoke_lambda(self.failover_lambda_name, payload_bytes)
+            if not failover_result:
+                raise CBCProxyException(f'Lambda failed for both {self.lambda_name} and {self.failover_lambda_name}')
+
+        return result
 
+    def _invoke_lambda(self, lambda_name, payload_bytes):
+        current_app.logger.info(f"Calling lambda {lambda_name}")
         result = self._lambda_client.invoke(
-            FunctionName=self.lambda_name,
+            FunctionName=lambda_name,
             InvocationType='RequestResponse',
             Payload=payload_bytes,
         )
+        current_app.logger.info(f"Finished calling lambda {lambda_name}")
 
         if result['StatusCode'] > 299:
-            raise CBCProxyException('Could not invoke lambda')
+            current_app.logger.info(
+                f"Error calling lambda {self.lambda_name} with status code { result['StatusCode']}, {result.get('Payload')}"
+            )
+            success = False
+
+        elif 'FunctionError' in result:
+            current_app.logger.info(
+                f"Error calling lambda {self.lambda_name} with function error { result['Payload']['errorType'] }"
```

Comment:
do we want to log the full payload here? Or is the expectation that we should then find the full lambda logs to figure out what went wrong? (that's fine too).

Reply:
Good point, we might as well log the full payload